### ISSUE TITLE
No way to cancel with specific error message

### ISSUE THREAD

**Issue by jedwards1211** [joeattardi/promise-poller#11]
I'm wanting to make a function that polls a GitHub commit status until my CI build has finished.  To do this with `promise-poller`, I have the following code:

```js
taskFn: () => this.repos.owner(owner).repo(repo).commits.ref(ref).status().then(({state}) => {
  if (state === 'failure') return false
  if (state === 'pending') throw new Error('commit is still pending')
}),
```

This is kind of backwards, at least for my use case, because I can't return a "build failed" error if `state === 'failure'`.  It would be better if I could return `false` when `state === 'pending'` and throw an error when `state === 'failure'`.

But really, considering that one might want to poll and eventually resolve to a boolean value, I think it would be best to support `isDone` and `shouldContinue` functions that default to the current behavior.

**Comment by joeattardi** [comment]
I mostly follow you and understand the use case. I have a couple of questions, though. The `isDone` and `shouldContinue` functions - would they be passed as options to the `promisePoller` call? What would they be called with?

**Comment by jedwards1211** [comment]
Yes, I was thinking they would be passed as options and get called with the `(err, result)` from each `taskFn` call.

**Comment by joeattardi** [comment]
I just added support for a `shouldContinue` function and published version 1.6.0. Let me know if this was not what you were looking for.

It takes two arguments - `err` and `result`. When a poll fails and the poll promise is rejected, `shouldContinue` will be called with the first argument set to the rejection reason. Then you can cancel further polling if you return `false` from this function. 

If the poll succeeds, it will be called with the second argument set to the resolved result. If you still want to poll despite getting a resolved value, you can return `true`.

If this doesn't solve your use case, feel free to re-open.
